workflow/validate/validate.go

```diff
@@ -214,15 +254,42 @@ func ValidateWorkflow(wftmplGetter templateresolution.WorkflowTemplateNamespaced
 	return wfConditions, nil
 }
 
+func ValidateFieldsWithWFTRef(v interface{}) error {
```

Comment:
Please choose a different name for this, perhaps?
```suggestion
func ValidateWorkflowSpecFields(v interface{}) error {
```